### PR TITLE
[Attention] add gpt-oss support

### PR DESCRIPTION
**Pull request by mikequan0425**
### What this PR does / why we need it?
Please refer to the following link for the historical conversation https://github.com/vllm-project/vllm-ascend/pull/4467. We have made updates in light of the comments from the prior PR review. Given the refactoring of the attention_v1 component, we have carried out necessary adjustments to fit the newly revised code.

### Does this PR introduce _any_ user-facing change?

1. Modified the code in the Attention section to adapt to the SWA and Sink features required by gpt-oss.
2. Modified the code in the MoE section to add support for bias and swigluoai.

### How was this patch tested?
Please refer to the https://github.com/vllm-project/vllm-ascend/pull/4467 for performance tests, on the basis of which the accuracy tests from AIME2024 have been newly added.
img_v3_02tu_501e88e3-2217-4565-8edf-b9acf4f43f2g

- vLLM version: v0.13.0
- vLLM main: https://github.com/vllm-project/vllm/commit/bde38c11df0ea066a740efe9b77fff5418be45df

### CODE REVIEW

File: vllm_ascend/attention/attention_v1.py [hunk 1]
```diff
@@ -352,6 +353,13 @@ def __init__(
         self.key_cache = None
         self.value_cache = None
         self.is_kv_producer = self.vllm_config.kv_transfer_config is not None and self.vllm_config.kv_transfer_config.is_kv_producer
+        self.device = None
+        if hasattr(torch, "npu") and callable(getattr(torch.npu, "is_available", None)) and torch.npu.is_available():
+            self.device = torch.device("npu")
+        else:
+            self.device = torch.device("cpu")
```
Comment (mikequan0425) [inline vllm_ascend/attention/attention_v1.py]: only npu

File: vllm_ascend/attention/attention_v1.py [hunk 2]
```diff
@@ -778,23 +781,54 @@ def forward_fused_infer_attention(
...
-        )
+        if self.sinks is not None:
+            actual_seq_qlen = attn_metadata.actual_seq_lengths_q
+            if attn_metadata.attn_state == AscendAttentionState.DecodeOnly:
+                actual_seq_qlen = torch.tensor([1] * len(attn_metadata.seq_lens_list), dtype=torch.int32).cumsum(dim=0)
+            # calculate atten_mask & sparse_mode by sliding_window
+            attn_mask_builder = AttentionMaskBuilder("npu")
+            if self.sliding_window is not None:
+                atten_mask = attn_mask_builder.get_swa_mask(torch.bool, self.sliding_window)
+                sparse_mode = 4
```
Comment (weijinqian0) [inline vllm_ascend/attention/attention_v1.py:790]: need move this to metadataBuilder.

Comment (mikequan0425) [reply vllm_ascend/attention/attention_v1.py:790]: Fixed, directly use the pre-constructed attention_mask in the metadata

**Comment by SlightwindSec** [comment]
Thanks for your contribution! It looks like the **DCO check failed**. Please follow this guide to fix it: https://docs.vllm.ai/projects/ascend/en/latest/developer_guide/contribution/index.html

**Comment by mikequan0425** [comment]
> Thanks for your contribution! It looks like the **DCO check failed**. Please follow this guide to fix it: https://docs.vllm.ai/projects/ascend/en/latest/developer_guide/contribution/index.html

@SlightwindSec Thank you for the reminder. I have completed the DCO check.
